src/include/daos_api.h

```diff
@@ -130,6 +142,19 @@ daos_tx_abort(daos_handle_t th, daos_event_t *ev);
 int
 daos_tx_close(daos_handle_t th, daos_event_t *ev);
 
+/**
+ * Restart the transaction handle. All modifications related with the handle
+ * will be dropped. This is a local operation, no RPC involved.
```

Comment:
More accurate to say: "This drops all modifications that have been issued with the handle."?

Also, do we need to promise this is a local, RPC-less operation?

Reply:
Good question. Currently, we cache modification on client, then restart will not touch RPC. We make the same 'promise' for TX abort API. I am not sure whether we will change related behavior in the future or not. But I think if we do not make such 'promise', we should adjust other TX API description also.

Reply:
In the public TX API documentation, I think it suffices to say, as a limitation, a TX cannot "see" its own uncommitted writes; it is unnecessary to say that we are buffering writes until commit time. If there's a significant benefit in revealing implementation details to applications, we could make an exception. That's just how I see it.

Reply:
The only reason i add such wording is to express that this operation should be fast; and does not need to be done asynchronously. We do that quite a bit in the API description, not just limited to the TX API. I can make a separate patch to remove all such comments from all APIs to address this concern.

Reply:
@mchaarawi, okay, I didn't know we do that quite a bit. We'd need a wider agreement before changing this then. Perhaps I'm the only one who has a concern. Thanks for the info.